lib/src/refescape.rs

```diff
@@ -0,0 +1,156 @@
+//! Escape strings for use in ostree refs.
```

Comment:
Now, perhaps this should go into https://github.com/ostreedev/ostree-rs - but that takes us down the new road of adding Rust-native API.  Which...well, we'll visit https://github.com/ostreedev/ostree/issues/2427 at some point.